pandas/tests/series/test_missing.py

```diff
@@ -1192,25 +1191,22 @@ def test_interp_limit_no_nans(self):
         expected = s
         assert_series_equal(result, expected)
 
-    def test_no_order(self):
-        tm._skip_if_no_scipy()
+    @td.skip_if_no_scipy
+    @pytest.mark.parametrize("method", ['polynomial', 'spline'])
+    def test_no_order(self, method):
         s = Series([0, 1, np.nan, 3])
         with pytest.raises(ValueError):
-            s.interpolate(method='polynomial')
-        with pytest.raises(ValueError):
-            s.interpolate(method='spline')
+            s.interpolate(method=method)
 
+    @td.skip_if_no_scipy
     def test_spline(self):
-        tm._skip_if_no_scipy()
         s = Series([1, 2, np.nan, 4, 5, np.nan, 7])
         result = s.interpolate(method='spline', order=1)
         expected = Series([1., 2., 3., 4., 5., 6., 7.])
         assert_series_equal(result, expected)
 
+    @td.skip_if_no('scipy', min_version='0.15')
     def test_spline_extrapolate(self):
-        tm.skip_if_no_package(
```

Comment:
As I'm adding comments I just realized that I can almost completely get rid of the ``tm.skip_if_no_package`` function as part of this change. It's practically always used used to check ``scipy`` (there's one check for ``pytables``) and could be replaced by the ``skip_if_no`` function in the ``_test_decorators`` module. Will clean up in next push